models/src_twitter_ads.yml

```diff
@@ -182,4 +188,92 @@ sources:
           - name: promoted_tweet_id
             description: The ID of the related promoted tweet.
           - name: url_clicks
-            description: The url clicks for the promoted tweet on that day.
\ No newline at end of file
+            description: The url clicks for the promoted tweet on that day.
+          - name: placement
+            description: Where on Twitter the ad is being displayed. Possible values include 'ALL_ON_TWITTER', 'PUBLISHER_NETWORK', 'TWITTER_PROFILE', 'TWITTER_SEARCH', 'TWITTER_TIMELINE', and 'TAP_*', which are more granular options for `PUBLISHER_NETWORK`.
+      - name: tweet
+        description: Each record represents a tweet, promoted or not. 
+        identifier: "{{ var('twitter_ads_tweet_identifier', 'tweet') }}"
+        columns:
+          - name: _fivetran_synced
+            description: "{{ doc('_fivetran_synced') }}"
+          - name: account_id
+            description: The ID of the related account.
+          - name: id
+            description: Unique identifier of the tweet.
+          - name: name 
+            description: If provided, the non-public title of the tweet. 
+          - name: full_text
+            description: Full text of the tweet's content.
+          - name: lang
+            description: Two-letter language code of the tweet. 
+      - name: campaign_report
+        description: Each record represents the performance of an ad campaign on a given day.
+        identifier: "{{ var('twitter_ads_campaign_report_identifier', 'campaign_report') }}"
+        columns:
+          - name: _fivetran_synced
+            description: "{{ doc('_fivetran_synced') }}"
+          - name: account_id
+            description: The ID of the related account.
+          - name: billed_charge_local_micro
+            description: The spend for the campaign on that day, in micros and in whichever currency was selected during account creation.
+          - name: clicks
+            description: The clicks for the campaign on that day.
+          - name: date
+            description: The date of the performance.
+          - name: impressions
+            description: The impressions for the campaign on that day.
+          - name: campaign_id
+            description: The ID of the ad campaign.
+          - name: url_clicks
+            description: The url clicks for the campaign on that day.
+          - name: placement
+            description: Where on Twitter the ad is being displayed. Possible values include 'ALL_ON_TWITTER', 'PUBLISHER_NETWORK', 'TWITTER_PROFILE', 'TWITTER_SEARCH', 'TWITTER_TIMELINE', and 'TAP_*', which are more granular options for `PUBLISHER_NETWORK`.
+      - name: line_item_report
+        description: Each record represents the performance of a line item (ad group) on a given day.
```

Comment:
I am a bit confused by this report. Are line items in Twitter essentially the equivalent of Ad Groups? 

I see in the [Ad Reporting Discovery doc](https://docs.google.com/document/d/1AvYasVocqqQ7h8C6WQ3ivslA707t4UTSzRwB3hTJYBw/edit#heading=h.ctmks52hja9n) you mention the hierarchy is 
```txt
Account > campaigns > ad groups > promoted tweets
```
Do line items fit into the ad group section?

Reply:
yep line items = ad groups